src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs

```diff
@@ -183,8 +184,36 @@ await using (SslStream tlsServer = new SslStream(serverStream))
 
                     if (revocationMode == X509RevocationMode.Offline)
                     {
-                        // Give the OCSP response a better chance to finish.
-                        await Task.Delay(200);
+                        if (offlineContext.GetValueOrDefault(false))
+                        {
+                            // Add a delay just to show we're not winning because of race conditions.
+                            await Task.Delay(200);
```

Comment:
From a white-box perspective, this delay doesn't actually do anything.  Because `offlineContext` is true the context isn't ever going to download an OCSP payload, so we're not helping the test pass or fail by delaying.

From a black-box/future-proofing perspective, the delay adds some confidence that the test isn't just passing by being faster than the product.

The intermittent failures are associated with the parameters that hit the else (that's now powered by reflecting to grab the task to ensure that we don't race ahead).